### DISCUSSION TITLE
Issue with permissions assigned via role

### DISCUSSION THREAD

**Discussion by joshcalloway** [spatie/laravel-permission#2766] [Q&A]
I recently hit a case where I had a user assigned to the role 'support' and the role had the 'support.manage' permission.  Any time I check with `can('support.manage')` or `hasPermissionTo('support.manage')` the check was always false even though the permission should be granted through the role.  I eventually "fixed" it by overriding the hasPermissionViaRole function on the User model with this code but I still don't see why it was failing to start with.
```
protected function hasPermissionViaRole(Permission $permission): bool
    {
        if (is_a($this, Role::class)) {
            return false; // Roles themselves do not inherit permissions from other roles
        }
        //this is the override below
        // Ensure roles are loaded and check if any of the user's roles match the permission's roles
        return $this->roles->contains(function (Role $role) use ($permission) {
            return $role->permissions->contains('id', $permission->id);
        });
    }
```

**Comment by parallels999** [comment]
maybe cache problem, reset permission cache
`php artisan permission:cache-reset`

**Reply by joshcalloway** [reply]
I did that so, so  many times.  I even changed cache providers, completely reinstalled redis, manually deleted the entire keyset for it.  It must have something to do with the way this is recursively called? `return $this->hasRole($permission->roles);`. Not sure.  Definitely not cache though.

**Reply by parallels999** [reply]
You could see if the cache is being saved correctly, that method is tested, `hasRole` supports Collections, and `$permission->roles` is a Role collection

Also maybe wrong guard
